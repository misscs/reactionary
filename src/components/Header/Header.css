header {
  border-bottom: 1px solid lightgrey;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.logo {
  margin-right: auto;
}
