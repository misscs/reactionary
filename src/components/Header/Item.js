import styled from 'styled-components'

const HeaderItem = styled.div`
  display: flex;
  border-left: 1px solid lightgrey;
  border-right: 1px solid lightgrey;
  padding: 0.5rem;
`

export default HeaderItem
