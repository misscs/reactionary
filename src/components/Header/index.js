import React, { PropTypes } from 'react'

import Badge from 'components/Badge'
import HeaderItem from './Item'
import './Header.css'

class Header extends React.PureComponent  {

  render () {

    const userActions = this.props.loggedIn ? (
        <HeaderItem>
          <Badge name={this.props.name} />
        </HeaderItem>
      ) : (
        <HeaderItem>
          Register
        </HeaderItem>

      )

    return (
      <header>
        <HeaderItem className="logo">
          <span>CC</span>
        </HeaderItem>

        { userActions }
      </header>
    )
  }
}


Header.propTypes = {
  loggedIn: React.PropTypes.bool.isRequired,
  isFetching: React.PropTypes.bool,
  userActions: React.PropTypes.node,
}


export default Header
