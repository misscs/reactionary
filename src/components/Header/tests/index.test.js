import React from 'react'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Header from 'components/Header'

describe('<Header />', () =>{

  it(`should only render the logo when logged out`, () => {
    const component = renderer.create(
      <Header
        loggedIn={false}
       />
    )

    const tree=component.toJSON()
    expect(tree).toMatchSnapshot()
  })


  it(`should render <Badge /> when logged in`, () => {
    const component = renderer.create(
      <Header
        loggedIn={true}
       />
    )

    const tree=component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
