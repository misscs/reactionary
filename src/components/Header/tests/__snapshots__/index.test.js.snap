// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should only render the logo when logged out 1`] = `
<header>
  <div
    className="logo cUhtom"
  >
    <span>
      CC
    </span>
  </div>
  <div
    className="cUhtom"
  >
    Register
  </div>
</header>
`;

exports[`<Header /> should render <Badge /> when logged in 1`] = `
<header>
  <div
    className="logo cUhtom"
  >
    <span>
      CC
    </span>
  </div>
  <div
    className="cUhtom"
  >
    <article
      className="badge"
    >
      <div
        className="avatar"
      />
      <div
        className="badge-content"
      />
    </article>
  </div>
</header>
`;
